modules/b2b_logic/entity_storage.c

```diff
@@ -412,9 +412,12 @@ static void receive_entity_create(enum b2b_entity_type entity_type,
 
 	return;
 error:
-	lock_release(&b2bl_htable[hash_index].lock);
```

Comment:
Moving this lock release after the `b2bl_delete()` function is the only change from these commits that has any effect. The rest of the changes are essentially NOPs.

Reply:
I tested only this change and it doesn't work on high load. What I offered works only for me. I debugged the code and crashes are because of structure not nulled.

Reply:
Maybe when you tested you had also other changes? Becuase setting `tuple = 0;` at the points in code where you did, only writes a local variable that is no longer read after that.

Reply:
I can't comment now because I investigated this for a long time ago,
but I tried to remove this few times and it definitely helped.
Now it's working with this patch in production and doesn't crash.

It only starts to affect on high CPS, let's say more than 400.
I had many crashes and this fixed for me.

If you think it's useless, I trust you. We'll see later how it works on latest b2b.

Reply:
I was just letting you know in case you wanted to update the PR so I could merge it. But I'm also ok with closing this PR and pushing the fix myself.

Reply:
Feel free to close it. 🙂